tests/perfalert/test_auto_perf_sheriffing/test_perf_sheriff_bot.py

```diff
@@ -101,6 +101,50 @@ def email_writer_mock():
         return MagicMock(spec=BackfillNotificationWriter())
 
 
+def test_records_change_to_ready_for_processing(
+    test_perf_alert,
+    create_record,
+    record_from_mature_report,
+    report_maintainer_mock,
+    backfill_tool_mock,
+    secretary,
+    sheriff_settings,
+    notify_client_mock,
+):
+    # create new report with records - the report will not be mature
+    create_record(test_perf_alert)
+
+    count_preliminary_records = BackfillRecord.objects.filter(
+        status=BackfillRecord.PRELIMINARY
+    ).count()
+    assert count_preliminary_records == 2
+    count_ready_records = BackfillRecord.objects.filter(
+        status=BackfillRecord.READY_FOR_PROCESSING
+    ).count()
+    assert count_ready_records == 0
+    count_frozen_reports = BackfillReport.objects.filter(frozen=True).count()
+    assert count_frozen_reports == 0
+
+    sheriff_bot = PerfSheriffBot(
+        report_maintainer_mock,
+        backfill_tool_mock,
+        secretary,
+        notify_client_mock,
+    )
+    sheriff_bot.sheriff(since=EPOCH, frameworks=['raptor', 'talos'], repositories=['autoland'])
+
+    count_preliminary_records_after = BackfillRecord.objects.filter(
+        status=BackfillRecord.PRELIMINARY
```

Comment:
Instead of duplicating this query, we could create it once at the start of the test function, and reuse it.
This could even make it easier to ditch the temporary variables from lines 120, 126 etc.